### ISSUE TITLE
"The semaphore timeout period has expired" Occuring more frequently

### ISSUE THREAD

**Issue by ogkimra** [Willy-JL/F95Checker#41]
![image](https://user-images.githubusercontent.com/32086892/199647670-464b9111-17b8-4a31-9c5a-d1a9874125ae.png)

**Comment by FaceCrap** [comment]
Since just posting a screenshot without any details, I'm curious if this happens regularly or if this was a one-off or incidental occurrence.

F95Zone regularly has a backup system running, if F95Checker happens to do a refresh in that period, I can imagine this happening.
I also have run into a situation where the whole site was inaccessible due to a database error. Which also would cause F95Checker to fail

I don't know if these situations will cause this exact error, because I only do manual refreshes, but it's important to at least add the following info:
- the refresh method used, 
- the time/date this happened (and which timezone you're in)

**Comment by Willy-JL** [comment]
> F95Zone regularly runs backups, if F95Checker happens to do a refresh in that period, I can imagine this happening.

That is true, but those are usually handled and F95Checker should show a "F95Zone backups are currently running, try again later" error message instead of the usual connection error. But of course if the message received from f95zone doesnt mention daily backups, even though the issue is daily backups, then it would not. However this error in particular seems to happen before the connection is established, no data is even received from f95zone, so telling the cause of the issue is trickier and likely its not on f95zone's side but on the client (unless its an f95zone outage / antispam protection).

However more information would indeed be useful

**Comment by Willy-JL** [comment]
I couldn't manage to reproduce the issue at all, but found some stuff online saying to just suppress the error so i did that and made it retry if it finds a WinError 121, try out [build 651](https://github.com/Willy-JL/F95Checker/actions/runs/3402653023) and let me know

**Comment by ogkimra** [comment]
> Since just posting a screenshot without any details, I'm curious if this happens regularly or if this was a one-off or incidental occurrence.
> 
> F95Zone regularly runs backups, if F95Checker happens to do a refresh in that period, I can imagine this happening. I also have run into a situation where the whole site was inaccessible due to a database error. Which also would cause F95Checker to fail
> 
> I don't know if these situations will cause this exact error, because I only do manual refreshes, but it's important to at least add the following info:
> 
>     * the refresh method used,
> 
>     * the time/date this happened (and which timezone you're in)
> 
>     * if the suggested workarounds were used and if that made a difference or not
> 
>     * number of workers in settings and the timeout value

![image](https://user-images.githubusercontent.com/32086892/200168682-e9129c3e-4f8e-431b-9c08-978229ebdf50.png)
1% to 97% is usually fast but it becomes very slow 98 onwards and they pops this error at 100%

I get the updates for games only after retrying 4-5 times

**Comment by Willy-JL** [comment]
As I said I can't manage to reproduce this issue. From what I can read online though it is a very obscure Windows error than often originates deeper in Windows hell, from either hardware or driver layers. Since to my knowledge only a limited amount of people are experiencing this issue (4 that I know of) I opted for a compromise. Default behavior is the same, but I added a toggle in refresh settings to ignore semaphore timeouts and WinError 121's. The note for that setting also mentions that it should be used as a last resort, and the preferred fix would be repairing the system with sfc, DISM and driver updates. [Build 655](https://github.com/Willy-JL/F95Checker/actions/runs/3405578543) contains this "fix", and it will be released in the next update unless a proper fix is found.
